pyflakes/checker.py

```diff
@@ -1699,7 +1699,34 @@ def CALL(self, node):
                     with self._enter_annotation():
                         self.handleNode(keyword.value, node)
 
-        self.handleChildren(node)
+        if any(_is_typing(node.func, nt, self.scopeStack)
+                for nt in ("TypedDict", "NamedTuple")):
+            with self._enter_annotation(AnnotationState.NONE):
+                self.handleChildren(node)
+
+            attributes = []
+
+            if node.func.id == "TypedDict":
+                if len(node.args) > 1 and isinstance(node.args[1], ast.Dict):
+                    attributes += node.args[1].values
+
+            elif node.func.id == "NamedTuple":
+                # NamedTuple("a", [("a", int)])
+                if len(node.args) > 1 and hasattr(node.args[1], "elts") and all(
+                    len(getattr(x, "elts", [])) == 2 for x in node.args[1].elts
+                ):
+                    attributes = [el.elts[1] for el in node.args[1].elts]
+
+                # NamedTuple("a", a=int)
+                attributes += [k.value for k in node.keywords]
+
+            for a in attributes:
+                if isinstance(a, ast.Str):
```

Comment:
I believe this is insufficient, consider `NamedTuple("T", ("a", List["Something"]))`

Reply:
This is already covered by [this](https://github.com/PyCQA/pyflakes/pull/609/files#diff-5df44d79406311387e0056faae7a8092d1ce9cc85c7d15420186e45a0c712283R1705). See [this test](https://github.com/PyCQA/pyflakes/pull/609/files#diff-a3d1d43d9902e26c2a2893ffbeb0fc7af7d94f35b39f8beaddd974770486ffb2R717)

Reply:
on your branch:

```python
from typing import NamedTuple
from t import A
from t import C

NamedTuple("A", ("a", A["C"]))
```

```
t.py:3:1 't.C' imported but unused
```

Reply:
I guess that would be expected, the forward reference only is checked if pyflakes knows the subscript caller is a member of a typing module. The same error is found with this :
```python
from typing import List
from t import C
A = List
b: A["C"]
```

Reply:
that doesn't produce an error -- pyflakes uses the knowledge of a "typing context" to process strings

Reply:
You're right. However it seems this error already exists in pyflakes current version. Should I improve this PR to manage this ?

Reply:
yes, you know which parts of the `NamedTuple` / `TypedDict` expression are typing contexts so you should process them with typing contexts instead of hacky `isinstance` of `str` -- that was my entire point for mentioning this

Reply:
My new push solves this by keeping a record of annotated nodes which are then always analysed inside a typing context